### PR TITLE
Handle clickhouse query exceptions in a graceful way

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: lib/sanbase/clickhouse_repo.ex [hunk 1]
```diff
@@ -54,23 +53,30 @@ defmodule Sanbase.ClickhouseRepo do
 
   defmacro query_transform(query, args, transform_fn \\ nil) do
     quote bind_quoted: [query: query, args: args, transform_fn: transform_fn] do
-      require Sanbase.ClickhouseRepo, as: ClickhouseRepo
+      try do
+        require Sanbase.ClickhouseRepo, as: ClickhouseRepo
+        require Logger
```
Comment (tspenov) [inline lib/sanbase/clickhouse_repo.ex]: Logger is not used

File: lib/sanbase_web/graphql/dataloader/clickhouse_dataloader.ex [hunk 2]
```diff
@@ -39,24 +46,47 @@ defmodule SanbaseWeb.Graphql.ClickhouseDataloader do
...
+      addresses
+      |> Enum.map(fn address ->
+        Map.get(eth_spent, address, {:ok, 0})
+      end)
+
+    project_eth_spent =
+      for({:ok, value} <- eth_spent_per_address, do: value)
+      |> Enum.sum()
+
+    has_errors? = Enum.any?(eth_spent_per_address, &match?({:error, _}, &1))
```
Comment (tspenov) [inline lib/sanbase_web/graphql/dataloader/clickhouse_dataloader.ex]: maybe more idiomatic

```elixir
eth_spent_per_address
|> Enum.any?(&match?({:error, _}, &1))
|> result(project_eth_spent)
```

File: lib/sanbase/blockchain/timescaledb.ex [hunk 3]
```diff
@@ -100,19 +100,24 @@ defmodule Sanbase.Timescaledb do
...
+
+          {:ok, result}
+
+        {:error, error} ->
+          {:error, error}
+      end
+    rescue
+      e ->
+        {:error, "Cannot execute TimescaleDB query. Reason: " <> Exception.message(e)}
     end
```
Comment (valo) [inline lib/sanbase/blockchain/timescaledb.ex]: It is really weird that the repo is throwing an exception instead of returning an error tuple.

Comment (IvanIvanoff) [reply lib/sanbase/blockchain/timescaledb.ex]: Depends. The functions without `!` can still raise if the error is not related to the function responsibilities. Not having a database/not being able to connect to db is example for such cases

**Pull request merged**